src/code/z_scene_proc.c

```diff
@@ -1,521 +1,804 @@
-        Actor_GetSwitchFlag(ctxt,0x35) &&
-        Actor_GetSwitchFlag(ctxt,0x36)) {
-        func_800C3C00(&ctxt->colCtx, 1);
-    } else {
-        func_800C3C14(&ctxt->colCtx, 1);
-    }
+    Gfx* dListHead;
+    Gfx* dList;
 
-    {
-        dl = (Gfx*)ctxt->state.gfxCtx->polyOpa.d - 18;
-        //dl = _g;
-        ctxt->state.gfxCtx->polyOpa.d = dl;
+    if (Flags_GetSwitch(globalCtx, 0x33) && Flags_GetSwitch(globalCtx, 0x34) && Flags_GetSwitch(globalCtx, 0x35) &&
+        Flags_GetSwitch(globalCtx, 0x36)) {
+        func_800C3C00(&globalCtx->colCtx, 1);
+    } else {
+        func_800C3C14(&globalCtx->colCtx, 1);
     }
 
-    SceneProc_DrawAllSceneAnimatedTextures(ctxt, ctxt->sceneTextureAnimations);
-
-    gfxCtx = ctxt->state.gfxCtx;
-    dlHead = dl;
-    for (i = 0; i < 9; i++, dlHead += 2) {
-        u32 lodFrac = 0;
-
-        bcopy(D_801C3C88, dlHead, sizeof(Gfx[2]));
-
-        switch(i) {
-        case 0:
-            if (Actor_GetSwitchFlag(ctxt,0x33) &&
-                Actor_GetSwitchFlag(ctxt,0x34) &&
-                Actor_GetSwitchFlag(ctxt,0x35) &&
-                Actor_GetSwitchFlag(ctxt,0x36)) {
-                lodFrac = 0xFF;
-            }
-            break;
-        case 1:
-            if (Actor_GetSwitchFlag(ctxt,0x37)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 2:
-            if (Actor_GetSwitchFlag(ctxt,0x37) &&
-                Actor_GetSwitchFlag(ctxt,0x38)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 3:
-            if (Actor_GetSwitchFlag(ctxt,0x37) &&
-                Actor_GetSwitchFlag(ctxt,0x38) &&
-                Actor_GetSwitchFlag(ctxt,0x39)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 4:
-            if (!Actor_GetSwitchFlag(ctxt,0x33)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 5:
-            if (Actor_GetSwitchFlag(ctxt,0x34)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 6:
-            if (Actor_GetSwitchFlag(ctxt,0x34) &&
-                Actor_GetSwitchFlag(ctxt,0x35)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 7:
-            if (Actor_GetSwitchFlag(ctxt,0x34) &&
-                Actor_GetSwitchFlag(ctxt,0x35) &&
-                Actor_GetSwitchFlag(ctxt,0x36)) {
-                lodFrac = 0x44;
-            }
-            break;
-        case 8:
-            if (Actor_GetSwitchFlag(ctxt,0x3A)) {
-                lodFrac = 0x44;
-            }
-            break;
+    dList = (Gfx*)GRAPH_ALLOC(globalCtx->state.gfxCtx, sizeof(Gfx) * 18);
+
+    Scene_DrawMaterialAnim(globalCtx, globalCtx->sceneMaterialAnims);
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    for (dListHead = dList, i = 0; i < 9; i++, dListHead += 2) {
+        lodFrac = 0;
+
+        bcopy(greatBayTempleColorSetDL, dListHead, sizeof(greatBayTempleColorSetDL));
+
+        switch (i) {
```

Comment:
`i` seems like it would be an enum. Looks like it's looping through display list slots. Probably to separate rendering of different surfaces in great bay? I would image this is what applies the shimmering water texture to everything.

Reply:
Yeah I didn't change this function very much from roz's original, and I agree that that makes more sense 👍

Reply:
Yeah, it seemed like you were doing a documentation update. That's a good time to improve this if you can get some good names.

Reply:
im honestly not sure what this does so idk how to rename it properly